tests/instantiate/__init__.py

```diff
@@ -2,13 +2,45 @@
 import collections
 import collections.abc
 from dataclasses import dataclass
+from functools import partial
 from typing import Any, Dict, List, Optional, Tuple
 
-from omegaconf import MISSING
+from omegaconf import MISSING, DictConfig, ListConfig
 
 from hydra.types import TargetConf
 
 
+def _convert_type(obj: Any) -> Any:
+    if isinstance(obj, DictConfig):
+        obj = dict(obj)
+    elif isinstance(obj, ListConfig):
+        obj = list(obj)
+    return obj
+
+
+def partial_equal(obj1: Any, obj2: Any) -> bool:
+    obj1, obj2 = _convert_type(obj1), _convert_type(obj2)
```

Comment:
Many places in this file there is the pattern `A == B or partial_equal(A, B)`.
This could be simplified by moving the `A == B` part inside of the `partial_equal` function as below:

```suggestion
def partial_equal(obj1: Any, obj2: Any) -> bool:
    if obj1 == obj2:
        return True

    obj1, obj2 = _convert_type(obj1), _convert_type(obj2)
```

This would enable simplifying the call-sites of `partial_equal`.
For example, the expression below on line 30
```python
            if not (obj1[i] == obj2[i] or partial_equal(obj1[i], obj2[i])):
                return False
```
could be replaced with this:
```python
            if not partial_equal(obj1[i], obj2[i]):
                return False
```

Reply:
I've tested out the change locally to make sure it works. If you agree with the idea, I can push a commit to this PR branch!

Reply:
nice!! feel free to push, thanks jasha!

Reply:
Done in https://github.com/facebookresearch/hydra/pull/1905/commits/bf689d30fd92d6f1fa46179202c9205df99e5a04